### PR TITLE
feat(auth): improve registration controls

### LINKED ISSUE #513: feat: 注册表单邮箱输入框右侧标注/下拉选择允许注册的邮箱域名，选中后缀后仅需填写邮箱前缀
## 反馈来源
QQ 群用户建议：注册/登录界面邮箱输入体验优化。

## 需求描述
在**注册表单**（以及可选的忘记密码表单）的邮箱输入框右侧分出一小段，展示当前**允许注册的邮箱域名白名单**：

1. 白名单只有 1 个域名时：直接在输入框右侧以固定后缀/标签形式展示（如 `@tongji.edu.cn`），用户只需填写邮箱前缀；
2. 白名单有多个域名时：右侧显示下拉菜单，用户先选择邮箱后缀域名，再只需填写前缀部分；
3. 选择后缀后自动拼接，避免用户输错域名（现后端会在域名不在白名单时拒绝，错误文案 `auth.emailDomain.notAllowed`：「该邮箱域名不在允许的注册白名单中」）。

## 现状（代码勘察）
- 注册/登录页：`apps/gooseforum/resource/src/site/pages/LoginPage.vue`（register 表单的邮箱输入为普通 `<input type="email">`，无域名提示，无后缀快捷选择）。
- 白名单配置：管理后台「注册与安全」→「允许注册的邮箱域名」（`SecurityAndRegistration.AllowedDomains`，见 `apps/gooseforum/app/models/forum/pageConfig/pageConfig.go`；**空 = 不限制**，管理端文案 `k0092`/`k0093`）。
- 后端校验：`apps/gooseforum/app/http/controllers/component/userUtil.go` 的 `ValidateEmailDomain`，注册/改邮箱都会校验域名白名单。
- `LoginPageProps`（`apps/gooseforum/app/http/controllers/forum/payload.go`）目前**未暴露** `AllowedDomains`，前端拿不到白名单，需要服务端下发。

## 建议实现要点
- 服务端：`LoginPageProps` 增加 `allowedDomains`（仅当白名单非空时有意义；为空表示不限制，前端不展示该 UI）。
- 前端：register 表单邮箱字段做成「前缀输入 + 后缀选择」组合控件：
  - 单个域名 → 直接固定后缀；
  - 多个域名 → 下拉选择后缀；
  - 为空 → 维持现状纯文本输入。
- 域名匹配后端 EqualFold（大小写不敏感），前端下发时建议统一小写。
- i18n 需覆盖 zh/en/ja/de 四套 locale。
- 备注：登录态本身不限域名（允许用户名或邮箱登录），本需求主要作用于注册表单；忘记密码表单（要求「注册邮箱」）可酌情一并支持。

### LINKED ISSUE #516: feat: 注册与安全新增「每日最大新用户数」限额（-1 不限），满员时注册页友好提示
## 需求描述

希望在管理面板「注册与安全」中新增「每日最大新用户数」设置项，用于限制每日新注册用户总量，防止激活邮件被批量刷发/恶意注册。

## 期望行为

1. **管理面板**：`AdminSettingsPage.vue` 的 security（注册与安全）段中，在「允许新用户注册」（k008y 开关）**下方**新增数值输入项「每日最大新用户数」：
   - 可填写数量，**-1 = 不限制**（建议作为默认值，保持现有行为不变）；
   - 附描述文案，例如「超过限额后当天暂停新用户注册，次日自动恢复」。
2. **注册接口拦截**：当今日新用户数达到限额时，`Register` 流程应在**用户创建及激活邮件发送之前**直接拒绝，返回友好错误（不得成功触发邮件发送）。
3. **注册页提示**：注册页看到友好提示「今日注册用户名额已满，请明天再来吧~」，样式沿用注册页现有错误提示（`LoginPage.vue` 中 `gf-status-message gf-status-message-error` 的 error 渲染样式）。

## 初步定位（静态扫描，供实现参考）

- 后端注册入口：`apps/gooseforum/app/http/controllers/api/authController.go` 的 `Register`（现有 `EnableSignup` 门禁之后、`userservice.CreateUser` / `emailactivationservice.SendActivationEmail` 之前插入限额检查即可满足「不触发邮件」）。
- 安全设置结构体：`apps/gooseforum/app/models/forum/pageConfig/pageConfig.go` 的 `SecurityAndRegistration`（新增字段如 `maxDailySignups`，默认 -1）。
- 默认配置：`apps/gooseforum/app/models/defaultconfig/pageconfig/security.json`。
- 保存/读取归一化：`adminController.go` 的 `SaveSecuritySettings`、`pageConfigRep.go`（参考 `maxDailyTopicsPerUser` 的负值归一化与缺失键回填默认值，issue #369 同款处理），必要时仿 `posting_settings.go` 增加数据迁移补默认键。
- 管理面板前端：`AdminSettingsPage.vue` security 段（`securityForm` + `normalizeSecurity`）、`src/admin/types.ts` 的 `SecuritySettings`；管理端文案需在 4 个 `admin-raw.*.generated.ts`（zh/en/ja/de）新增键。
- 消息码：`apps/gooseforum/app/http/controllers/component/message_code.go` 新增（如 `auth.register.dailyQuota`），并在站点 i18n `locales/zh.ts`（`server.auth.register` 下）及 en/ja/de 补充文案，注册页经现有 `errorMessage`/`resolveApiMessage` 机制即可渲染，无需改前端展示逻辑。

## 计数口径与边界建议

- 「当日」沿用服务器本地时区口径（`time.Now().Format("2006-01-02")`），与 `topics.CantWriteNew` / 上传每日限制一致。
- 计数即 `users` 表当日新建行数（`created_at` 已有索引），新增 COUNT 查询即可；count 后 insert 非原子属可接受的软性防滥用护栏（与 issue #369 语义相同，不强一致）。
- 需确认 0 的语义：按本需求 -1 = 不限，则 0 可视为「当天不允许注册」（次日自动恢复）；也可选择归一化为不限。建议实现时明确并在管理端描述文案中写清。

## 验收标准

1. 默认 -1 时注册行为与现状完全一致。
2. 设置 N>0 且当日注册数达到 N 后，再次注册返回上述友好提示；不创建账号、不发送激活邮件；次日恢复。
3. 管理端可正确保存/回显该配置（含 -1 与负数归一化边界）。
4. 新增 i18n 键 4 语言齐全，管理端与注册页文案正常显示。

### PR DESCRIPTION
**Pull request by WALKERKILLER**
## Summary / 摘要

完善注册流程的两个安全与体验能力：

- 注册邮箱白名单在前端直接体现为邮箱后缀选择，减少用户手动输入完整学校邮箱时的出错概率；
- 增加每日新用户注册软限额，由管理端配置，并在达到上限时给注册用户明确、友好的提示。

本 PR 同时保持既有后端邮箱域名校验规则、蜜罐静默行为和单一二进制部署方式不变。

Closes #513
Closes #516

### 注册邮箱域名选择（#513）

- 登录页 SSR payload 新增 `allowedDomains`，直接复用安全设置中的邮箱域名白名单。
- 当白名单为空时，注册页保持原有完整邮箱输入框，不改变现有行为。
- 当仅配置 1 个域名时：
  - 注册页固定显示 `@域名` 后缀；
  - 用户仅填写邮箱前缀；
  - 提交前在前端拼接为完整邮箱。
- 当配置多个域名时：
  - 使用现有 `SiteSelect` 组件提供后缀下拉选择；
  - 用户仅填写邮箱前缀；
  - 提交前按当前选择拼接完整邮箱。
- 登录、找回密码仍接受完整邮箱，不受此交互调整影响。
- 后端 `ValidateEmailDomain` 继续作为最终校验来源，白名单为空时不限域名，域名比较仍大小写不敏感。

### 每日注册软限额（#516）

- 安全设置新增 `maxDailySignups`：
  - `-1`：不限制，作为默认值，升级后保持原有注册行为；
  - `0`：当天禁止创建新用户；
  - 正整数：达到对应当天新用户数量后暂停继续注册。
- 限额按服务器本地日历日统计 `users.created_at`，次日自然恢复，不额外引入计数表、缓存或后台任务。
- 限额判断位于蜜罐静默处理之后，避免改变既有反机器人响应行为。
- 达到上限时返回稳定 message code `auth.register.dailyQuota`，站点四语言均提供友好提示，不再退化为通用“注册失败”。
- 管理端“安全与注册”增加每日最大新用户数配置项。
- 对历史 `securitySettings` 配置做兼容读取：缺少新字段时自动使用默认 `-1`，避免旧配置被错误解释为 `0`。
- 保存或读取 `< -1` 的非法值时归一为 `-1`。

## Verification / 验证

本地实际执行并通过：

- [x] `git diff --check`
- [x] 相关 Go package tests（auth / forum payload / message code / admin route contract / page config / users / config cache）
- [x] `cd apps/gooseforum && go vet ./...`
- [x] `scripts/hooks/check-golangci.sh`：`0 issues.`
- [x] `cd apps/gooseforum/resource && pnpm run check:i18n`
- [x] `cd apps/gooseforum/resource && pnpm typecheck`
- [x] `cd apps/gooseforum/resource && pnpm test`：82 个测试文件、644 个测试通过
- [x] `cd apps/gooseforum/resource && pnpm build`
- [x] `cd packages/api-contract && pnpm run check`
  - OpenAPI lint / bundle 通过；
  - 1078 个外部 fixture 校验通过；
  - TypeScript 生成产物无漂移；
  - route coverage：293 routes = 231 covered + 0 known-uncovered + 62 excluded。
- [x] lefthook pre-commit：staged whitespace / gofmt 通过
- [x] lefthook pre-push：go vet / golangci-lint / frontend typecheck 通过

说明：当前执行环境没有可用 PostgreSQL 服务且无 Docker socket 权限，因此仓库要求的 PostgreSQL migration gate 无法在本机启动；本 PR 触及模型读取层但未新增/修改表结构或 migration，仍由 CI 的 `ci-backend-pg` 使用真实 PostgreSQL 环境完成最终门禁。

OpenAPI lint 仍报告仓库已有的 13 条“operation 缺少 4XX response” warning，本 PR 未新增该类 warning。

## Docs & contract impact / 文档与契约影响

- 更新 `docs/product/current-state.md`：记录注册邮箱域名选择和每日注册软限额的当前行为。
- `AdminSecuritySettingsConfig` OpenAPI 契约新增 `maxDailySignups`：
  - `type: integer`
  - `minimum: -1`
  - `default: -1`
- 同步提交生成的 Web TypeScript OpenAPI 类型。
- 更新管理员安全设置 contract fixture 与 route-level HTTP contract test。
- 注册页 SSR `LoginPageProps` / Web client payload 类型同步新增 `allowedDomains`。
- 中文、英文、日文、德文站点提示和管理端文案保持一致。

## Known gaps / 已知缺口

- 本地未运行 PostgreSQL migration gate，等待 CI `ci-backend-pg` 验证。
- 每日注册限额是刻意设计的“软限额”：使用注册前的实时 COUNT，不增加锁、计数器表或分布式原子计数；极端并发下允许少量越界，符合本 issue 的轻量限流定位。

### CODE REVIEW

File: apps/gooseforum/resource/src/admin/pages/AdminSettingsPage.vue [hunk 1]
```diff
@@ -406,6 +407,7 @@ function normalizeSecurity(settings: Partial<SecuritySettings> = {}) {
   return {
     enableSignup: toBool(settings.enableSignup, true),
     enableEmailVerification: toBool(settings.enableEmailVerification, false),
+    maxDailySignups: Number(settings.maxDailySignups ?? -1),
```
Comment (oierxjn) [inline apps/gooseforum/resource/src/admin/pages/AdminSettingsPage.vue]: **blocker**：`Number('') === 0`。`<Input v-model.number type="number">` 清空输入框后 `.number` 修饰符回退为空字符串 `''`，而 `?? -1` 只挡 nullish 不挡空串 → 这里得到 `0`，保存即「当日全站禁止注册」，全程无提示。后端 `SaveSecuritySettings` 只钳 `< -1`，`0` 是合法值救不了；而清空是管理端最自然的「取消限制」手势。

建议一行修复（顺带镜像后端 `< -1` 钳位）：

```ts
const n = Number(settings.maxDailySignups)
maxDailySignups: Number.isFinite(n) && n >= -1 ? Math.trunc(n) : -1,
```

其中 `n >= -1` 分支只是前端镜像后端规则（行为一致、回显友好），核心是 `''`/NaN → `-1`，别落进 `0`。

File: apps/gooseforum/app/models/forum/users/users_rep.go [hunk 2]
```diff
@@ -216,6 +216,15 @@ func ExistEmail(email string) bool {
 	return builder().Select("1").Where("email = ?", email).Limit(1).Scan(&id).RowsAffected > 0
 }
 
+// CountCreatedToday 返回当天新建用户数量，用于注册软限额。
+func CountCreatedToday() int64 {
```
Comment (oierxjn) [inline apps/gooseforum/app/models/forum/users/users_rep.go:220]: **should**：限额核心分支缺直接单测——本函数、`GetSecuritySettingsConfig` 的兼容读取（存量配置缺字段→-1 / 显式 null→默认 / `< -1` 归一 / 坏 JSON→默认值告警）、`Register` 的限额拒绝路径（`0` 边界、`N` 边界、`auth.register.dailyQuota` code）均无测试；目前只有 admin 契约测试碰到 `-1` 和 `7` 两个值。

建议：`GetSecuritySettingsConfig` 表驱动测试（旧配置 / 显式 null / 负值 / 坏 JSON 四分支）+ 注册限额 HTTP 测试（预插 N 个当日用户，断言拒绝与 message code）；本函数可顺带覆盖当日起点边界。

File: apps/gooseforum/app/http/controllers/api/adminController.go [hunk 3]
```diff
@@ -1442,6 +1442,9 @@ type SaveSecuritySettingsReq struct {
 
 // SaveSecuritySettings 保存安全与注册设置
 func SaveSecuritySettings(req component.BetterRequest[SaveSecuritySettingsReq]) component.Response {
+	if req.Params.Settings.MaxDailySignups < -1 {
```
Comment (oierxjn) [inline apps/gooseforum/app/http/controllers/api/adminController.go:1445]: **nit**：顺带一提，本函数下方第 1452 行读 `current` 仍用旧读取器 `GetConfigByPageType`，而同文件 1435 行（GetSecuritySettings）已切 `GetSecuritySettingsConfig`。`current` 仅用于禁用用户名 diff、无功能影响，但同一配置两套读取路径易漂移，建议统一。

File: apps/gooseforum/resource/src/site/pages/LoginPage.vue [hunk 4]
```diff
@@ -168,6 +171,7 @@ async function handleRegister() {
     error.value = t('auth.validation.registerRequired')
     return
   }
+  const email = page.props.allowedDomains.length ? `${registerForm.email}@${selectedEmailDomain.value}` : registerForm.email
```
Comment (oierxjn) [inline apps/gooseforum/resource/src/site/pages/LoginPage.vue:174]: **nit**：受限模式下前缀框允许粘入含 `@` 的完整邮箱，会拼出 `a@b@c` 提交；后端 `ValidateEmailDomain` 的 `len(parts) != 2` 会拒绝（fail-closed，无安全风险），纯 UX 瑕疵。可在提交前对前缀 strip 一次 `@`。

File: apps/gooseforum/resource/src/locales/ja.ts [hunk 5]
```diff
@@ -150,6 +150,8 @@ export default {
     usernameOrEmail: 'ユーザー名またはメール',
     username: 'ユーザー名',
     email: 'メール',
+    emailPrefix: 'メールの前半',
```
Comment (oierxjn) [inline apps/gooseforum/resource/src/locales/ja.ts]: **nit**：「メールの前半」（“前半”）表意不明确，建议「メールのローカル部分」或「@より前の部分」。可留。

**Review by oierxjn — CHANGES_REQUESTED** [review]
## Review 结论：Request Changes

1 blocker + 1 should + 3 nit，行内评论见上。实现与契约同步整体质量很高，修完 B1（一行）即可合入。

### B1（blocker）`AdminSettingsPage.vue:410`
管理端清空「每日最大新用户数」输入框 → `Number('') === 0` → 落库 `0` = 当日全站禁注，静默、无提示。`< -1` 在本 PR 两条路径都会归一为 `-1`（保存钳位 + 读取 `max(x, -1)`），唯独空串产生的 `0` 绕过所有防线且是最危险的合法值。

### S1（should）限额核心分支缺直接单测
`CountCreatedToday` / `GetSecuritySettingsConfig` 兼容读取四分支 / `Register` 限额拒绝路径（`0` 边界、`N` 边界、dailyQuota code）均无直接测试，目前只有 admin 契约测试碰到 `-1` 和 `7`。

### 已核对无问题
- 安全：限额检查在蜜罐之后、建用户之前；`allowedDomains` SSR 暴露是 #513 既定产品语义；无迁移/schema 变更；`users.created_at` 有索引，COUNT 为索引范围扫描，软限额成本可接受（PR 已声明取舍）。
- 契约：schemas/fixture/生成 TS 同步，本地重跑 `generate:ts` 无漂移；移动端无此配置引用，无需 Dart 镜像；无新路由。
- 前端：域名 0/1/N 三态渲染有测试；SiteSelect 用法匹配；i18n 四语言齐全。
- 文档：`current-state.md` Auth 行已更新。

### 本地已跑（review 工作树 @ 5f06ef0f）
`go build ./...` + `go vet ./...` ✅；admin security settings 契约测试 8/8 ✅；`pnpm typecheck` ✅；`vitest auth-brand-theme.test.ts` 13/13 ✅；`check:i18n` ✅；`generate:ts` 无 diff ✅。PG 迁移门禁未本地跑（本 PR 无迁移变更，留给 CI `ci-backend-pg`）。

**Review by yzxoi — DISMISSED** [review]
Verified locally and approving. Both halves are solid:

**Email domain selector (#513):**
- SSR payload carries `allowedDomains` straight from the security-settings cache — single source of truth, no drift from the admin-configured whitelist. The three UI states are all covered: empty list keeps the exact old full-email input (type=email preserved), single domain renders a fixed `@domain` suffix (no dropdown to misclick), multiple domains reuse `SiteSelect`. Full email is composed only at submit time; login/forgot flows untouched; `ValidateEmailDomain` remains the backend authority (case-insensitive semantics unchanged).
- Compose-bug check: the suffix is appended to the raw trimmed prefix, and the prefix input no longer has `type=email` in restricted mode (a full email pasted in can't double-compose since users type only the prefix — and a stray `@` in the prefix still fails backend validation, fail-closed).

**Daily signup soft quota (#516):**
- Correct semantics on all three values: `-1` unlimited (default → upgrade is a no-op), `0` pauses the day, positive caps. Gate sits after the honeypot silent-accept (existing anti-bot response shape unchanged) and before user creation/activation email.
- The compat read is the key correctness detail: `GetSecuritySettingsConfig` probes the stored JSON for `maxDailySignups` and falls back to the default `-1` when the field is absent — without this, every pre-upgrade stored config would have been decoded as `0` and silently disabled registration. Clamp `max(x, -1)` on read plus the save-side clamp closes malformed values from both ends.
- `CountCreatedToday` uses server-local midnight on `created_at` with no new table/cache/job — right size for a soft quota, and the next day self-heals.
- Stable `auth.register.dailyQuota` message code with all four locales + the admin i18n keys; contract three-piece synced (`AdminSecuritySettingsConfig` required field + regenerated openapi.ts + fixture); mobile Dart mirrors don't carry this schema (verified — no sync needed); docs updated in-PR (Auth row).

**Verification:** local — api/routes/users/pageConfig/hotdataserve test suites all pass (admin settings contract covers the new field round-trip); new LoginPage tests cover the single-suffix / multi-domain compose / empty-list regression; typecheck + CI fully green including PG migration suite (pageConfig shape) and race detector. Merging after #511 so the import graph is settled first.

File: apps/gooseforum/resource/src/admin/pages/AdminSettingsPage.vue (hunk 1 above)
Comment (WALKERKILLER) [reply apps/gooseforum/resource/src/admin/pages/AdminSettingsPage.vue]: 已修复：空字符串/空白/NaN 现在都会归一为 -1；有限数值会截断并钳到 >= -1，避免清空输入误落库为 0。见 e3782e52。

File: apps/gooseforum/app/models/forum/users/users_rep.go (hunk 2 above)
Comment (WALKERKILLER) [reply apps/gooseforum/app/models/forum/users/users_rep.go:220]: 已补核心分支直测：GetSecuritySettingsConfig 覆盖缺字段/null/<-1/显式值/坏 JSON，CountCreatedToday 覆盖本地日历日边界，注册 HTTP 覆盖 0 和正数边界并断言 auth.register.dailyQuota 且不创建用户。见 e3782e52。

File: apps/gooseforum/app/http/controllers/api/adminController.go (hunk 3 above)
Comment (WALKERKILLER) [reply apps/gooseforum/app/http/controllers/api/adminController.go:1445]: 已统一为 GetSecuritySettingsConfig，避免同一安全配置在保存流程出现两套读取语义。见 e3782e52。

File: apps/gooseforum/resource/src/site/pages/LoginPage.vue (hunk 4 above)
Comment (WALKERKILLER) [reply apps/gooseforum/resource/src/site/pages/LoginPage.vue:174]: 这点暂不改：受限模式下含 @ 的前缀继续由后端 ValidateEmailDomain fail-closed 拒绝。自动 strip/改写用户输入会引入额外容错语义，当前行为更显式，也不影响安全性。

File: apps/gooseforum/resource/src/locales/ja.ts (hunk 5 above)
Comment (WALKERKILLER) [reply apps/gooseforum/resource/src/locales/ja.ts]: 已改为更明确的「@より前の部分」。见 e3782e52。

**Comment by oierxjn** [comment]
@WALKERKILLER 合并一下最新dev

**Review by oierxjn — APPROVED** [review]
## 复审通过（Approve）

复审 @ 4d6eec00（含 e3782e52 修复提交 + dev 同步/markdown2html 搬迁的机械 import 修正），首轮评审的 1 blocker + 1 should + 3 nit 全部妥善处理：

- **B1 已修复**：`normalizeSecurity` 现在把空串/空白/NaN 归一为 `-1`，`Math.max(-1, Math.trunc(n))` 同时镜像了后端 `< -1` 钳位并消除小数导致 Go int 解码失败的问题；`0`（合法的「当日禁注」）正确保留。逐一核对了 `''` / 空白串 / null / `-1` / `0` / `7` / `-9` / `1.9` / NaN 各分支，语义与后端一致。
- **S1 已补齐**：`TestGetSecuritySettingsConfigCompatibility`（缺字段/显式 null/负值/坏 JSON/显式限额 5 分支表驱动）、`TestRegisterHTTPContractDailySignupQuota`（`0` 全拒 + 限额 1 的当日边界，均断言 message code 且未创建用户）、`TestCountCreatedToday`（昨日/今日零点边界）。
- **N1** 读取器已统一为 `GetSecuritySettingsConfig`；**N3** 日语译名已改「@より前の部分」；**N2**（前缀 strip `@`）作为可留 nit 未处理，可接受。
- e3782e52 之后的两个提交仅为同步 dev + `markdown2html` 包搬迁的 import 路径机械修正，不触及本 PR 功能面。

### 复审已跑（review 工作树 @ 4d6eec00）
`go build ./...` + `go vet ./...` ✅；`TestGetSecuritySettingsConfigCompatibility` 5/5 ✅；`TestCountCreatedToday` ✅；`TestRegisterHTTPContractDailySignupQuota` 2/2 ✅；admin security settings 契约测试回归 ✅；`pnpm typecheck` ✅；`check:i18n` ✅。PG 门禁仍由 CI `ci-backend-pg` 兜底。

非阻塞备注：`normalizeSecurity` 的新归一化逻辑无前端单测（函数内嵌在 SFC 中，与该页既有测试模式一致），后续如重构 admin 设置页可顺手抽纯函数补测。
